autogen/agentchat/conversable_agent.py

```diff
@@ -633,12 +672,23 @@ def generate_oai_reply(
         if messages is None:
             messages = self._oai_messages[sender]
 
+        # unroll tool_responses
+        all_messages = []
+        for message in messages:
+            tool_responses = message.get("tool_responses", [])
+            if tool_responses:
+                all_messages += tool_responses
+                # tool role on the parent message means the content is just concatentation of all of the tool_responses
```

Comment:
Much of these implicit assumptions here regarding message schema should be eventually refactored into a centralized class.

Reply:
Maybe in another PR, but the way we currently doing it will cause many bugs in the future.

Reply:
Yup, this whole agent is ripe for types and multiple class refactors :-)